resources/js/components/shared/TimelinePlot.vue

```diff
@@ -0,0 +1,393 @@
+<template>
+  <div id="timeline_plot"></div>
+</template>
+<script>
+export default {
+  name: "TimelinePlot",
+  props: {
+    plotData: {
+      type: Object,
+      required: true,
+      default: () => { return null }
+    },
+  },
+  mounted() {
+    this.plot_timeline_graph();
+  },
+  methods: {
+    plot_timeline_graph() {
+      /*
+       * This template generates a line plot for a group of time-series datasets,
+       * where each set is allowed to have a different number of data points.
+       *
+       * The expected input is of the form:
+       *   plotData = {
+       *   | data: [
+       *   |   {
+       *   |   | color: <line color>,  <--- d3 accepts most color formats (hex, rgb, etc.)
+       *   |   | name: "<name to appear in legend>",
+       *   |   | values: [
+       *   |   |   {
+       *   |   |   | x: <Date object>,
+       *   |   |   | y: <numeric>,
+       *   |   |   | url: "<url>"      <--- optional, render clickable points if provided
+       *   |   |   },
+       *   |   |   ...
+       *   |   | ],
+       *   |   },
+       *   |   ...
+       *   | ],
+       *   | labels: {
+       *   |   title: "<plot title>",
+       *   |   x_axis: "<x-axis label>",
+       *   |   y_axis: "<y-axis label>"
+       *   | }
```

Comment:
I don't think it makes sense to put this in the `plotData` prop.  We can reasonably assume that there is only one of these for each timeline plot, so I think it would make more sense to make each of these a separate prop.  In fact, I think it would make sense to eliminate the `plotData` prop entirely, and have the following props: `data`, `title`, `xAxis`, and `yAxis`.  Doing so would reduce the amount of validation work, and allow Vue to automatically handle default values for you.

Reply:
Good idea. Will do